### PR TITLE
Add: in-memory OpenAI Files API (/v1/files)

### LINKED ISSUE #71: File Upload
Hi,

I am following the documentation for setup open-webui and N8N.
I have enabled vision and file transfer, disabled file context.
The chat and images are sucessfully reaching  N8N but pdf nor word and other files types are not being transferred.

i am using the latest version of open-webui [v0.8.10]
My docker compose setup is a s follows:
  n8n-openai-bridge:
    image: ghcr.io/sveneisenschmidt/n8n-openai-bridge:latest
    container_name: n8n-openai-bridge
    ports:
      - "3334:3333"
    environment:
      - BEARER_TOKEN=xxxxx
      - N8N_WEBHOOK_BEARER_TOKEN=
      - LOG_REQUESTS=true
      - SESSION_ID_HEADERS=X-Session-Id,X-Chat-Id,X-OpenWebUI-Chat-Id
      - REQUEST_BODY_LIMIT=250mb
      - FILE_UPLOAD_MODE=extract-multipart
      - USER_ID_HEADERS=X-User-Id,X-OpenWebUI-User-Id
      - USER_NAME_HEADERS=X-User-Name,X-OpenWebUI-User-Name
      - ENABLE_TASK_DETECTION=true
    volumes:
      - ./models.json:/app/models.json:ro
    restart: unless-stopped

### PR DESCRIPTION
**Pull request by sveneisenschmidt**
## Summary

- Implements OpenAI `/v1/files` endpoints with in-memory storage and TTL-based expiration (fixes #71)
- Resolves file_id references in chat completion messages to inline base64 data URLs before forwarding to n8n
- Adds shared MIME type utility, unified env int parser, and renames `WebhookNotifier` → `WebhookNotifierService` for consistency

## New endpoints

| Method | Path | Description |
|--------|------|-------------|
| `POST` | `/v1/files` | Upload file (multipart/form-data) |
| `GET` | `/v1/files` | List files with filtering/pagination |
| `GET` | `/v1/files/:file_id` | Get file metadata |
| `GET` | `/v1/files/:file_id/content` | Download file content |
| `DELETE` | `/v1/files/:file_id` | Delete file |

## Configuration

| Env Var | Default | Description |
|---------|---------|-------------|
| `FILES_ENABLED` | `true` | Enable/disable Files API |
| `FILES_MAX_FILE_SIZE` | 20MB | Max size per file |
| `FILES_MAX_TOTAL_STORAGE` | 200MB | Max total in-memory storage |
| `FILES_TTL_SECONDS` | 3600 | File expiration time |
| `FILES_CLEANUP_INTERVAL_SECONDS` | 60 | Cleanup sweep interval |

## Manual testing

Update your existing installation to use the PR image tag:

```
ghcr.io/sveneisenschmidt/n8n-openai-bridge:pr-72
```

Add the new environment variables to your configuration (see table above) and restart.

Test the new `/v1/files` endpoints and file attachments in chat completions with your existing setup.

## Test plan

- [x] 76 test suites, 716 tests passing
- [x] 96%+ coverage maintained
- [x] Lint and format clean
- [ ] Manual test: upload, list, retrieve, delete via curl
- [ ] Integration test: upload file, reference in chat completion, verify n8n receives content
- [ ] Open WebUI: attach PDF/DOCX, verify file reaches n8n workflow

### CODE REVIEW

File: docs/CONFIGURATION.md [hunk 1]
```diff
@@ -392,7 +392,46 @@ FILE_UPLOAD_MODE=passthrough         # Options: passthrough, extract-json, extra
...
+|----------|---------|-------------|
+| `FILES_ENABLED` | `true` | Enable/disable the `/v1/files` endpoints. Set to `false` to disable entirely. |
+| `FILES_MAX_FILE_SIZE` | `20971520` (20MB) | Maximum size per uploaded file in bytes. |
+| `FILES_MAX_TOTAL_STORAGE` | `209715200` (200MB) | Maximum total in-memory storage for all files. |
+| `FILES_TTL_SECONDS` | `3600` (1 hour) | Time-to-live for stored files before automatic eviction. |
+| `FILES_CLEANUP_INTERVAL_SECONDS` | `60` | How often the cleanup sweep runs to remove expired files. |
+
+**Endpoints:**
+- `POST /v1/files` - Upload a file (multipart/form-data)
+- `GET /v1/files` - List uploaded files
```
Comment (tobias-gp) [inline docs/CONFIGURATION.md:422]: since the API is not tailored towards specific users, individual LibreChat users could list all files. Should this be default behaviour being able to list all available files? Or would requiring to know the UUID introduce sufficient "pseudo-security"? 

(hope I got it right ;), otherwise correct)

File: src/repositories/FileRepository.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+  /**
+   * List files with optional filtering and pagination
+   * @param {Object} [options={}]
+   * @param {string} [options.purpose] - Filter by purpose
+   * @param {number} [options.limit=10000] - Max results
+   * @param {string} [options.order='desc'] - Sort order by created_at ('asc' or 'desc')
+   * @param {string} [options.after] - Cursor: return files after this ID
+   * @returns {Array<Object>} Array of file metadata objects
+   */
+  list({ purpose, limit = 10000, order = 'desc', after } = {}) {
```
Comment (tobias-gp) [inline src/repositories/FileRepository.js:90]: Maybe protect by env var, otherwise return empty list?

File: src/utils/mimeTypes.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  'image/svg+xml': 'svg',
+  'application/pdf': 'pdf',
+  'text/plain': 'txt',
+  'application/json': 'json',
+  'application/msword': 'doc',
+  'application/vnd.openxmlformats-officedocument.wordprocessingml.document': 'docx',
+  'application/vnd.ms-excel': 'xls',
+  'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet': 'xlsx',
+  'text/csv': 'csv',
+};
```
Comment (tobias-gp) [inline src/utils/mimeTypes.js:40]: add PPTX?

**Review by tobias-gp — APPROVED** [review]
as per the comments in the code my only concern would be privacy across multiple users.

**Comment by tobias-gp** [comment]
## Security Review: Files API

Reviewed the changes introduced in this branch (`feature/files-api`). Found three issues and implemented fixes in 7376294.

### Findings & Fixes

| # | Severity | Issue | Fix |
|---|----------|-------|-----|
| 1 | **HIGH** | Content-Disposition header injection via unsanitized filename — raw `req.file.originalname` used directly in `attachment; filename="..."` header | `FileService.sanitizeFilename()` strips path components, control chars, and `"` / `\` before storage |
| 2 | **MEDIUM** | No filename validation at upload time — path traversal patterns, control characters, and arbitrary-length filenames stored as-is | Same `sanitizeFilename()` applied in `upload()`, enforces 255-char max with extension preservation |
| 3 | **MEDIUM** | No file count limit — authenticated user could exhaust memory via millions of tiny files (Map overhead >> file bytes, bypassing `FILES_MAX_TOTAL_STORAGE`) | New `FILES_MAX_COUNT` config (default: 1000), enforced in `upload()` with `FILE_COUNT_LIMIT_EXCEEDED` error |

### Additional observations (not addressed, lower priority)

- **Files API enabled by default** — `FILES_ENABLED` defaults to `true` (opt-out). Consider opt-in for reduced attack surface on upgrade.
- **Shared rate limiter** — file uploads use the `standard` limiter (100 req/min). A dedicated, stricter limiter for `POST /v1/files` would reduce memory pressure from upload bursts.
- **Cached multer singleton** — `filesMaxFileSize` is captured once at first request. Requires restart to change (no hot-reload). Minor, but worth documenting.

### Test coverage

15 new tests covering:
- `sanitizeFilename`: path traversal (Unix/Windows), control chars, quotes, truncation, edge cases (null, empty, dots, whitespace)
- File count limit: enforcement, error code, recovery after deletion
- Route: 413 response for count limit, Content-Disposition header safety assertion
- Config: `FILES_MAX_COUNT` default, parsing, invalid/zero value handling

**Comment by sveneisenschmidt** [comment]
thanks @tobias-gp, I added an env var to have listing off by default. research has shown this is not actively used by librechat and openwebui

**Comment by sveneisenschmidt** [comment]
I haven't heard back from the person that raised the issue. Would love to see some people testing this first too.
